File: app/src/main/java/com/ahmedadeltito/photoeditor/RegistrationActivity.java
package com.ahmedadeltito.photoeditor;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Spinner;
import android.widget.TextView;

public class RegistrationActivity extends AppCompatActivity {

    private EditText txtName;
    private EditText txtMobileNumber;
    private EditText txtEmail;
    private EditText txtPassword;
    private EditText txtAddress;
    private Spinner spnGender;
    private TextView lblDateOfBirth;
    private Button btnRegister;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_registration);

        txtName = (EditText) findViewById(R.id.txtName);
        txtMobileNumber = (EditText) findViewById(R.id.txtPhonerNumber);
        txtEmail = (EditText) findViewById(R.id.txtEmail);
        txtPassword = (EditText) findViewById(R.id.txtPassword);
        txtAddress = (EditText) findViewById(R.id.txtAddress);
        spnGender = (Spinner) findViewById(R.id.spnGender);
        lblDateOfBirth = (TextView) findViewById(R.id.lblDateOfBirth);
        btnRegister = (Button) findViewById(R.id.btnRegister);

        btnRegister.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if(validate()){

                }else {

                }
            }
        });


    }

    private boolean validate() {



        return false;
    }
}

File: app/src/main/java/com/ahmedadeltito/virtualdressingview/RegistrationActivity.kt
package com.ahmedadeltito.virtualdressingview

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.support.design.widget.Snackbar
import kotlinx.android.synthetic.main.activity_registration.*
import org.json.JSONObject

class RegistrationActivity : AppCompatActivity(), OnResultListener {

    private lateinit var progressDialogUtil: ProgressDialogUtil

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_registration)
        progressDialogUtil = ProgressDialogUtil(this)

        btnSignUp.setOnClickListener {
            init()
        }
        btnCancel.setOnClickListener {
            finish()
        }
    }

    private fun init() {
        progressDialogUtil.show()
        val registrationAsync = RegistrationAsync()
        var request = JSONObject()
        var param = JSONObject()
        param.put("Name", txtName.text)
        param.put("Username", txtUserName.text)
        param.put("Password", txtPassword.text)
        param.put("Phone", txtNumber.text)
        param.put("Type", "")
        request.put("request", param)
        registrationAsync.setOnResultListener(this, request)
    }


    override fun onSuccess(result: String?) {
        progressDialogUtil.hide()
        if (!result.isNullOrBlank()) {
            val snackbar = Snackbar.make(btnSignUp, "Registration completed.", Snackbar.LENGTH_INDEFINITE)
            snackbar.setAction("Ok") {
                finish()
            }
            snackbar.show()
        } else {
            val snackbar = Snackbar.make(btnSignUp, "An error has occurred.try later", Snackbar.LENGTH_LONG)
            snackbar.setAction("Ok") {

            }
            snackbar.show()
        }
    }
}
